### PR TITLE
SIDM-8461-I want to use a PIN

### PR DESCRIPTION
**Pull request by chandkanuri**
### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/app/pin-service/PinService.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    return this.idamApiAxios
+      .get(apiUrl, {
+        headers: {
+          pin,
+          'Content-Type': 'application/json',
+        },
+      })
+      .catch(error => {
+        this.logger.error('Error logging with pin from IDAM API');
```
Comment (jburke-idam) [inline src/main/app/pin-service/PinService.ts:22]: "logging"?

File: src/main/app/pin-service/PinService.ts [hunk 2]
```diff
@@ -9,18 +8,18 @@ export class PinService {
   constructor(private readonly idamApiAxios: AuthorizedAxios, private readonly logger: Logger) {}
 
   public loginWithPin = (clientId: string, redirectUri: string, pin: string) => {
-    const apiUrl = this.PIN_ENDPOINT + '?client_id=' + clientId + '&redirect_uri=' + redirectUri;
-
     return this.idamApiAxios
-      .get(apiUrl, {
+      .get(this.PIN_ENDPOINT + '?client_id=' + clientId + '&redirect_uri=' + redirectUri, {
```
Comment (kremi) [inline src/main/app/pin-service/PinService.ts:12]: Shouldn't we be  passing the parameters in a "safer"  way, eg
```suggestion
      .get(this.PIN_ENDPOINT, {
          params: {
               client_id: clientId,
               redirect_uri:  redirectUri
          }
      }, {
```

Comment (kremi) [reply src/main/app/pin-service/PinService.ts:12]: Also - there  might be an optional "state" parameter that we need to pass to idam-api if it's been sent to us.

File: src/test/unit/test/app/pin-service/PinService.test.ts [hunk 3]
```diff
@@ -24,35 +23,5 @@ describe('PinService', () => {
 
       expect(pinService.loginWithPin(clientId, redirectUri, pin)).resolves.toBe(true);
     });
-
-    test('Should throw error if error from axios', () => {
```
Comment (kremi) [inline src/test/unit/test/app/pin-service/PinService.test.ts]: Why  are we removing those tests?

Comment (chandkanuri) [reply src/test/unit/test/app/pin-service/PinService.test.ts]: Code clean up, Added new tests.

**Review by kremi — APPROVED** [review]

**Pull request merged**